lib/kubeclient.rb

```diff
@@ -12,16 +12,14 @@ module Kubeclient
   class Client
     attr_reader :api_endpoint
 
-    ENTITY_TYPES = %w(Pod Service ReplicationController Node Event Endpoint
-                      Namespace)
-
     # Dynamically creating classes definitions (class Pod, class Service, etc.),
     # The classes are extending RecursiveOpenStruct.
     # This cancels the need to define the classes
     # manually on every new entity addition,
     # and especially since currently the class body is empty
-    ENTITY_TYPES.each do |entity_type|
-      Object.const_set(entity_type, Class.new(RecursiveOpenStruct))
+    ENTITY_TYPES = %w(Pod Service ReplicationController Node Event Endpoint
+                      Namespace).map do |et|
+      [Kubeclient.const_set(et, Class.new(RecursiveOpenStruct)), et]
```

Comment:
No reason.  We could do a hash too, but I didn't see a place where we needed to do a hash lookup.